File: 백준/Silver/2559. 수열/수열.py
import sys
input = sys.stdin.readline

n,k = map(int,input().split())
lst = list(map(int,input().split()))


s = sum(lst[:k])
answer =s

for i in range(1,n-k+1):
    s = s-lst[i-1]+lst[i+k-1]
    answer = max(answer,s)
print(answer)

File: 백준/Silver/2491. 수열/수열.js
const fs = require("fs");
// TODO: 제출 시 경로 변환 필수 ("/dev/stdin")
//5.37//5.53
let input = fs.readFileSync("/dev/stdin").toString().trim().split("\n");
const n = Number(input.shift());
let lst = input[0].split(" ").map(e => Number(e));

let plus = 1;
let minus = 1;
let answer = 0;
for (let i = 1; i < lst.length; i++) {
  if (lst[i] >= lst[i - 1]) plus++;
  else if (plus > 0) {
    answer = Math.max(answer, plus);
    plus = 1;
  }
  if (lst[i] <= lst[i - 1]) minus++;
  else if (minus > 0) {
    answer = Math.max(answer, minus);
    minus = 1;
  }
}

answer = Math.max(answer, minus);
answer = Math.max(answer, plus);
console.log(answer);
